### DISCUSSION TITLE
Trust Registry Queries

### DISCUSSION THREAD

**Discussion by guru-barbes** [trustoverip/tswg-trust-registry-tf#72] [Show and tell]
This discussion thread is to help focus the collection and collating of the queries and claims that a trust registry should be able to respond to.

I've started a googledoc here: https://docs.google.com/spreadsheets/d/1tZmAlq__GW_3-fCA6XkIE8qJXg8D4nNhF_ujm_Q4kTk/edit?usp=sharing which attempts to organise the various queries and claims.

Please add anything you think should be included - add a tab or reformat, etc to make this useful to everyone!

**Comment by darrellodonnell** [comment]
v1.0 supports 3 questions:
* Is this {DID} an Authoritative Issuer for {credential type} under {EGF}
* Is this {DID} an Authorized Verifier for {presentation type} under {EGF}
* Is this other {DID} a recognized Trust Registry under {EGF}

Further questions include:
* What roles are recognized in the {EGF}? 
  * NOTE: to me this needs to be as simple as possible lest we end up creating an RBAC.
* What DID methods are supported under the {EGF}?
* What schema definitions are supported under the {EGF}?
* What credential formats are supported under the {EGF}?
* What credential types (schema, cred defs as required) are supported under the {EGF}
* What revocation approaches are in use under the {EGF}?
* What overlays are used under the {EGF}?
* What presentation requests are available under the {EGF}?
  * refine by adding "...to which {role}?"
* What wallets are authorized under the {EGF}?
* more...

**Reply by darrellodonnell** [reply]
looking at the following in the Google Sheet, I think we need to back up. The "source of truth" is really an ecosystem governance framework (EGF). The governance section questions are hard to answer without going circular. 
<img width="404" alt="image" src="https://user-images.githubusercontent.com/686855/218494471-0d86256f-215e-48cf-8ca3-0c26b1377397.png">


Some of the information is what I call metadata. I have added commentary to the Google Sheet.


A single Trust Registry endpoint MAY support/host many EGFs so some questions are really hard to answer. The root of things is the DID of the EGF.

**Comment by darrellodonnell** [comment]
Another thought,related to "claims" - the claims below seem to lean towards the use of VCs as the chain of provenance. While that's a great goal, I see it as quite far away. Individual ecosystems need answers now to their own internal questions. The simpler we can make those answers available, the better. 


<img width="482" alt="image" src="https://user-images.githubusercontent.com/686855/218495987-03af0316-87be-43fa-8201-f18f43573ad7.png">

**Reply by guru-barbes** [reply]
Thanks Darrell for the great list of questions (I've added them to the Googledoc) and the helpful comments to. Looking forward to discussing and tightening up this list of queries and logic.
